### ISSUE TITLE
Loosing dynamic functions when using VueJS

### ISSUE THREAD

**Issue by jmbluethner** [dattran92/site-prerender#2]
Hey @dattran92 👋

I've managed to set up a CI/CD pipeline for my VueJS project.
At first I thought that everyting is running fine, but after testing my page I found out, that all of my dynamic functions are now lost.
I can't close my cookie consent, some buttons aren't working and so on.

I think the problem is that site-prerender doesn't "translate", or parse, the JavaScript from the VueJS components.
Also, VueJS doesn't find it's root element (which makes sense because the Vue container element gets replaced by the actually rendered components).

TL;DR: Puppeteer doesn't seem to care about my JavaScript.

Cheers

**Comment by dattran92** [comment]
@jmbluethner Could you check your code or provide a simple version that we could look into. 

The example (https://blog.dattran.one) I'm using is working correctly.

- You can see in the screenshot that the root element is still there: 
![image](https://user-images.githubusercontent.com/8003687/165413634-36b1605a-1334-4069-a7ff-5d1e9f5e4743.png)

- Client side routing works normally in my case, which means the JS works correctly.

**Comment by jmbluethner** [comment]
Hi @dattran92 

Thanks for your reply!

I've created a super simplified version of my page. Feel free to get it from https://gitlab.com/jmbluethner/site-prerender-test
When you run the page from the /src folder, you will see a popup which can be dismissed. However, after rendering the page, using <code>npm run render</code>, you can't click on the popup anymore.
Btw, I'm not using any routing. I just use Vue to create re-usable, dynamic components.

I've read that you have to run puppeteer in the non-headless mode in order to render the JS correctly. So I've tried that by chaning line 8 in <code>/lib/crawler.js</code> (in site-prerender) from <code>this.headless = headless || true</code> to <code>this.headless = headless || false</code>. But this just resulted in some Node / Express errors while crawling. The page was still being rendered to <code>/dist</code> though.

**Comment by dattran92** [comment]
Hi @jmbluethner Sorry for late response. I'm not sure if it is because of your VueJS version or your setup, but the problem is that the generated HTML of your vue application doesn't contain the root `<div id='app'></div>`, which makes the client side unable to be executed. 

You can take a look at my example here to see if this could help you to adjust your project setup to make it works.
https://github.com/dattran92/site-prerender-vue-example

So as this is not the problem of the package itself. I'll close the issue now.
